core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -573,7 +574,24 @@ func (ec *EthConfirmer) batchFetchReceipts(ctx context.Context, attempts []EthTx
 		}
 
 		if receipt.Status == 0 {
-			l.Warnf("transaction %s reverted on-chain", receipt.TxHash)
+			// Do an eth call to obtain the revert reason.
+			_, errCall := ec.ethClient.CallContract(ctx, ethereum.CallMsg{
```

Comment:
Is there any value doing this async? Not sure if there are requirements on how long `batchFetchReceipts` could take but in the worst case doing an eth_call for a bunch of reverted tx's might add quite a bit of overhead.

Reply:
I'd say mass reversions are indicative of a bug (not sure of any scenario in which thats normal) so fine to do it inline imo